### PR TITLE
Add dtests for LWT with static columns and batches

### PR DESCRIPTION
**Pull request by ifesdjeen**
Related Cassandra ticket is [9842](https://issues.apache.org/jira/browse/CASSANDRA-9842)

### CODE REVIEW

File: cql_tests.py [hunk 1]
```diff
@@ -798,3 +798,50 @@ def materialized_view_test(self):
                                     retry_policy=FallthroughRetryPolicy())
         assert_unavailable(lambda c: debug(c.execute(statement)), session)
         node2.watch_log_for("operations timed out", from_mark=mark, timeout=60)
+
+
+class LWTTester(CQLTester):
+    @since('2.1')
+    def lwt_with_static_columns_test(self):
```
Comment (ptnapoleon) [inline cql_tests.py]: Can you add the following docstring?
```
"""
@jira_ticket CASSANDRA-9842
"""
```

File: cql_tests.py [hunk 2]
```diff
@@ -798,3 +798,50 @@ def materialized_view_test(self):
                                     retry_policy=FallthroughRetryPolicy())
         assert_unavailable(lambda c: debug(c.execute(statement)), session)
         node2.watch_log_for("operations timed out", from_mark=mark, timeout=60)
+
+
+class LWTTester(CQLTester):
```
Comment (ptnapoleon) [inline cql_tests.py]: What's the need for a new subclass? Is there a reason this wasn't added to unit tests?

Comment (ifesdjeen) [inline cql_tests.py]: It was added to the unit tests. Problem is that there're two code paths for LWT (via `executeWithCondition` and via `executeInternalWithCondition`, the former one only happens during the network requests, so I decided to put it here. 

I've added similar test for the internal code path to the unit tests. 

There was no reason for the new class though.

File: cql_tests.py [hunk 3]
```diff
@@ -798,3 +798,53 @@ def materialized_view_test(self):
...
+        res = session.execute("BEGIN BATCH\n" +
+                              "INSERT INTO lwt_with_static (a, b, d) values (4, 4, 'a');\n" +
+                              "UPDATE lwt_with_static SET s = 4 WHERE a = 4 IF s = null;\n" +
+                              "APPLY BATCH;")
+        self.assertEqual(rows_to_list(res), [[True]])
+
+        res = session.execute("SELECT * FROM lwt_with_static WHERE a = 4")
+        self.assertEqual(rows_to_list(res), [[4, 4, 4, "a"]])
+
+        print res
```
Comment (ptnapoleon) [inline cql_tests.py]: Why `print res`?

**Comment by ptnapoleon** [comment]
Do we want to run this test on a cluster with more than one node?

**Comment by ifesdjeen** [comment]
@ptnapoleon for LWT-yes. There's a bit more info here: https://issues.apache.org/jira/browse/CASSANDRA-9842?focusedCommentId=15277770&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15277770

File: cql_tests.py [hunk 4]
```diff
@@ -798,3 +798,51 @@ def materialized_view_test(self):
                                     retry_policy=FallthroughRetryPolicy())
         assert_unavailable(lambda c: debug(c.execute(statement)), session)
         node2.watch_log_for("operations timed out", from_mark=mark, timeout=60)
+
+    @since('2.1')
+    def lwt_with_static_columns_test(self):
+        """
+        @jira_ticket CASSANDRA-9842
+        """
+        session = self.prepare()
```
Comment (ptnapoleon) [inline cql_tests.py]: You'll want to set `nodes=3` here, as a kwarg to prepare

**Comment by ptnapoleon** [comment]
In which case, you'll want to create a 3 node cluster in your prepare statement. +1 after that.

**Comment by ptnapoleon** [comment]
Bump, the prepare still needs updated, and I see this is waiting on the commit of the ticket

**Comment by ifesdjeen** [comment]
Yes, sorry for not updating it. Decided to wait until the dependency ticket gets merged, which only happened yesterday. Will get to it today.

**Comment by ifesdjeen** [comment]
I'll add a couple more tests, as noted by @blerer. I'll leave a note when it's done here.

**Comment by ifesdjeen** [comment]
@blerer I've added some more tests, you think it's sufficient?

**Comment by blerer** [comment]
The tests look good to me.

**Comment by ptnapoleon** [comment]
@ifesdjeen youre good to merge. There are some minor style nits, from travis, but i'll have @c-kodman clean those up, as I need him to make changes to the assertions youre using for a PR currently in flight anyway

**Comment by ifesdjeen** [comment]
I've fixed Travis complaints! Didn't notice that one before.
Merged.

**Comment by ifesdjeen** [comment]
Thank you!

**Pull request merged**